actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -750,10 +769,45 @@ func findByChannelLookup(ctx context.Context, storage exec.Storage, byPayer exec
 
 // checkCondition combines params in the condition with the redeemerSuppliedParams, sends a message
 // to the actor and method specified in the condition, and returns an error if one exists.
-func checkCondition(vmctx exec.VMContext, condition *types.Predicate, redeemerSuppliedParams []interface{}) (uint8, error) {
+func checkCondition(vmctx exec.VMContext, chid *types.ChannelID, payerAddress address.Address, condition *types.Predicate, redeemerSuppliedParams []interface{}) (uint8, error) {
 	if condition == nil {
 		return 0, nil
 	}
+
+	if len(redeemerSuppliedParams) == 0 {
```

Comment:
A comment explaining that this is the cached condition would help readers a lot.

Reply:
sry cached condition is overloaded, I mean the situation in which `checkCondition` will look in its cache